### PR TITLE
media-gfx/maim-3.4.47 add new package

### PR DESCRIPTION
**Pull request by mrVanDalo**
maim (Make Image) is a utility that takes screenshots of your desktop
using imlib2. It's meant to overcome shortcomings of scrot and
performs better in several ways.

I created [Bug 599362](https://bugs.gentoo.org/show_bug.cgi?id=599362) as described in the [wiki](https://wiki.gentoo.org/wiki/Gentoo_Github)

### CODE REVIEW

**Comment by mrVanDalo** [comment]
Ups :smile: will fix that

**Comment by mrVanDalo** [comment]
Actually I can't figure out how `x11-misc/slop` came into the `funtoo` portage tree.

File: media-gfx/maim/maim-3.4.47.ebuild [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+inherit cmake-utils
+
+DESCRIPTION="takes screenshots of your desktop is designed to be an improved scrot"
```
Comment (SoapGentoo) [inline media-gfx/maim/maim-3.4.47.ebuild]: Titlecase "takes"

File: media-gfx/maim/Manifest [hunk 2]
```diff
@@ -0,0 +1,3 @@
+DIST maim-3.4.47.tar.gz 46799 SHA256 fb12524cc05f7f4ff46fe40b951334c6dbc274e50b6d0ea08185f555ca3cd74d SHA512 4b21f8ad8f9f2f19693d5c489f5e815133e1b8308653c52934cdb8a3d8ae38732733a5ffb5527b44db49f59c3818a9bc83840ced5db71807a68277d0dcc03392 WHIRLPOOL de21bb02c2923fd158cc691df09ff9a55121206f2f491a6b4bcf080b7b56032822f7ac4bad98028294f5f6e6e103fb393db3933dc3a7efcd383db6602d2d17d9
+EBUILD maim-3.4.47.ebuild 557 SHA256 dd2828950e6b4f89fa0feceaa965dfd15cdb3b50c00c802a0c9858bf9b2c6d53 SHA512 294753541b6f6264120afb9d30c972ddeb2aa4a0294707f1aef7294cfbfe4626eb78e8ab1545eafc5fc4f648ca7942b1723ab5ac80484bb4ed0e9894fd1f4492 WHIRLPOOL e7080c98dbe30e8e8a8a958d02145f6a6871cf851d07564b55dba600aa36e84a47f12531b27e680bcd8b8902a07021a43a29a63a66975ce2aded45711756ff8b
+MISC metadata.xml 559 SHA256 2fb690bc6edf9ec1be0f5a8341159e0b1c9a8251eb3031841650e111e289267f SHA512 9204e50059992d04ef1d117c74621bbf4e4d20299a3e8af97483aa7f36daff18dc6015d9656845ff60085d737597d6b5e3a1e490ce60e292f906fd6e399e2af7 WHIRLPOOL 04b1bd5adb2b4dde97b50e830186fbfb4391a82a186ff0e3d515bcc2c5f2211ca0eaa8e48f170c73922148504547fcfc64969b58ca92a0b07b27f8d1dd8d86cd
```
Comment (SoapGentoo) [inline media-gfx/maim/Manifest]: this is a fat manifest, the normal gentoo tree uses thin manifests. Please fix these

File: media-gfx/maim/metadata.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person">
+		<email>gentoo@ingolf-wagner.de</email>
+		<description>Primary maintainer</description>
+	</maintainer>
```
Comment (SoapGentoo) [inline media-gfx/maim/metadata.xml]: you need to add the proxy maintainers project: https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide#How_proxy_maintainership_works

**Comment by mrVanDalo** [comment]
@SoapGentoo Thanks for the hint, I'll fix them as soon as possible, which not that quick right now, because my internet situation is quite bad for the next 2 weeks. But I'll fix it for sure!

**Comment by mrVanDalo** [comment]
I will get an internet connection tomorrow (I hope) so I will have some time this weekend to make this PR straight

**Comment by mrVanDalo** [comment]
Ok, lets get some work done :D

**Comment by mrVanDalo** [comment]
@SoapGentoo hope you like my changes.

I checked it on a docker container which runs a gentoo.
Also checked repoman before.
If there is still something, please let me know.
It's my first PR for upstream, and I want to make it proper.

**Comment by mrVanDalo** [comment]
Hmm while I'm looking at the change list (of slop).
I don't think this is a good patch.
I should at least make it configurable.

Sorry it's late, I'll have a look at it tomorrow.

File: x11-misc/slop/slop-4.3.21.ebuild [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
```
Comment (SoapGentoo) [inline x11-misc/slop/slop-4.3.21.ebuild]: can you port this to `EAPI=6` please?

File: x11-misc/slop/slop-4.3.21.ebuild [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="opengl"
+
+RDEPEND="
+	x11-libs/libX11
+	x11-libs/libXext
+
+	opengl? (
+		media-libs/glew:*
```
Comment (SoapGentoo) [inline x11-misc/slop/slop-4.3.21.ebuild]: glew has subslots, if it links to glew, you need to use `:0=` instead

File: x11-misc/slop/slop-4.3.21.ebuild [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+		media-libs/glm
+	)
+"
+
+src_prepare() {
+	epatch "${FILESDIR}"/no-cppcheck.patch
+	epatch "${FILESDIR}"/no-gengetopt.patch
+	#Apparently librt is necessary to compile for Ubuntu
+	#It isn't needed or marked as a dependency here so be sure it isn't compiled with it
+	epatch "${FILESDIR}"/no-librt.patch
```
Comment (SoapGentoo) [inline x11-misc/slop/slop-4.3.21.ebuild]: add these 3 to the global `PATCHES` array

File: x11-misc/slop/slop-4.3.21.ebuild [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+	use opengl || epatch "${FILESDIR}"/no-opengl.patch
```
Comment (SoapGentoo) [inline x11-misc/slop/slop-4.3.21.ebuild]: `use opengl || PATCHES+=( "${FILESDIR}"/no-opengl.patch )`

File: x11-misc/slop/slop-4.3.21.ebuild [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+}
```
Comment (SoapGentoo) [inline x11-misc/slop/slop-4.3.21.ebuild]: and don't forget to call `cmake-utils_src_prepare` at the end!

File: x11-misc/slop/slop-4.3.21.ebuild (hunk 5 above)
Comment (mrVanDalo) [inline x11-misc/slop/slop-4.3.21.ebuild]: I have to admit I don't really get the difference of Slots and SubSlots, but maybe glew is a bad example, I wrote me a todo to figure that out someday soon.

Comment (mrVanDalo) [inline x11-misc/slop/slop-4.3.21.ebuild]: But this is fixed now, I used `:0=`

**Comment by mrVanDalo** [comment]
Hey @SoapGentoo I fixed the issues you had. Hope it's fine now.

**Comment by bpinto** [comment]
@mrVanDalo Given you have removed `gengetopt` from slop (using a patch). Do you think it makes sense to remove from `maim` too?

**Comment by mrVanDalo** [comment]
@bpinto good  point. Will do that.

**Pull request closed without merging**